hcn/hcnnetwork.go

```diff
@@ -415,15 +415,22 @@ func GetNetworkByID(networkID string) (*HostComputeNetwork, error) {
 		return nil, err
 	}
 	hcnQuery.Filter = string(filter)
+	queryJson, err := json.Marshal(hcnQuery)
+	if err != nil {
+		return nil, err
+	}
 
-	networks, err := ListNetworksQuery(hcnQuery)
+	networkGuid, err := guid.FromString(networkID)
 	if err != nil {
 		return nil, err
 	}
-	if len(networks) == 0 {
-		return nil, NetworkNotFoundError{NetworkID: networkID}
+
+	network, err := getNetwork(networkGuid, string(queryJson))
```

Comment:
>getNetwork [](http://example.com/codeflow?start=17&length=10)

what's the difference between enumerating the networks and return the first one and retrieve the network by ID?

If this is necessary, do we need to change all GetNetworkByxxx?

Reply:
Enumerating networks will get the list of available networks at that point. Since DefaultSwitch is a special System-Created switch whose is triggered on HcnOpenEndpoint("DefaultSwitch") call... So just calling EnumerateNetworks will not return DefaultSwitch unless HcnOpenEndpoint("DefaultSwitch") is called...

Reply:
need any other information?